### ISSUE TITLE
a11y - edit page - skip links - cms ui

### ISSUE THREAD

**Issue by Wagner3UB** [plone/volto#5268]
UPDATE 25/10/2023: This issue was modified to align the content with the landmark modifications created by David, something I was not aware of until I saw the pull request. The skip links are a useful tool to navigate, but aren't the only solution to navigate between important "points of interests". The skip links must work inside the CMS-UI, this is important, and this was done.

**Describe the bug**
This issue will tackle two problems, essentially, but it can serve as a solution for other accessibility situations in the future:

1. When you click the 'edit' button, the default behavior is to focus on the first editable block, which is usually the title block. However, when using assistive technology, this behavior can be inconsistent, leaving users unsure of their location and what action to take. To improve this, the fieldset containing all blocks should be made accessible to screen readers and provide clear information about the editable content before moving to the first element. - **Another issue (https://github.com/plone/volto/pull/5392) to solve this problem**
2. Accurate link identification is crucial for a smooth navigation experience. This means that link names and anchor points should align with the editing page's content. The current "footer/navigation/content" labels aren't entirely accurate and should instead reference terms like "edit main content," "toolbar configurations," or "editable content sidebar."

**To Reproduce**

1- Go to https://demo.plone.org/edit
2- enter in edit mode by clicking on the edit button (top-left)
3- `shift + tab` and navigate to the skip links "skip to footer", "skip to main content" and "skip to navigation"

**Expected behavior**
The expected behavior needs to be discussed, and various options are available. I will provide some previous information, but there are many ways to set this behavior.

- ~~[ ] Make the first focus more informative for assistive technologies~~ - **Another issue (#5392) to solve this problem**
- [ ] The skip links within the CMS-UI can serve as a 'point of interest and redirect.' This means that if the user needs to change their screen position (e.g., from the sidebar to the main content or from the toolbar to the sidebar), they can do so by navigating to the skip links and finding what they need with a proper label ("edit main content," "toolbar configurations," or "editable content sidebar."). This will work to improve the UX and give the user another useful way to access those parts of the CMS-UI besides using the landmarks.
- ~~[ ] Skip links should be accessible early on. I see two possibilities: either the focus shifts to the first skip link after clicking the 'Edit' button or, even before that, when users access the CMS (after logging in), the first click on the edit page should activate a skip link~~. The skip links can be accessed via landmarks
- [x] The second option brings us to another scenario. The CMS's main page doesn't have a sidebar, which means the skip links should also be different from the 'edit' page. - With the update made by David the landmark to arrive inside the sidebar have a label "Sidebar empty complementary", which means there's nothing to see there.
- [x] Skip links must be accessible from any part of the page, ensuring that the page never enters an infinite loop at any time.

**Screenshots**
<img width="794" alt="Schermata 2023-10-04 alle 10 02 04" src="https://github.com/plone/volto/assets/60133113/9e8b377d-a0b5-47c0-be2d-d73465775362">

**Comment by Wagner3UB** [comment]
@sneridagh @davisagli @stevepiercy @pnicolli I will need your help to determine the direction to be followed in this issue. I believe it involves UX/UI, and a strategy should be taken into consideration.

**Comment by Wagner3UB** [comment]
To manage the skiplinks issue: https://github.com/plone/volto/issues/5117
